### PR TITLE
Use float<->half conversion routines from the OpenCL headers

### LINKED ISSUE #870: Use the float<->half conversion routines from the headers
Conversion routines have been added to the headers by https://github.com/KhronosGroup/OpenCL-Headers/pull/60. We need to use them instead of the code in https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_common/harness/imageHelpers.cpp#L898

### PR DESCRIPTION
**Pull request by kpet**
Fixes #870

Signed-off-by: Kevin Petit <kevin.petit@arm.com>

### CODE REVIEW

File: test_conformance/half/Test_vStoreHalf.cpp [hunk 1]
```diff
@@ -208,211 +210,27 @@ CheckD(cl_uint jid, cl_uint tid, void *userInfo)
...
-    return (u.u >> (24-11)) | sign;
+static cl_ushort float2half_rtn( float f ) {
+    return cl_half_from_float(f, CL_HALF_RTN);
 }
 
+static cl_ushort double2half_rte( double f );
+static cl_ushort double2half_rtz( double f );
+static cl_ushort double2half_rtp( double f );
+static cl_ushort double2half_rtn( double f );
+
```
Comment (jrprice) [inline test_conformance/half/Test_vStoreHalf.cpp:233]: What's the reason for not using `cl_half_from_double` for these ones?

Comment (kpet) [reply test_conformance/half/Test_vStoreHalf.cpp:233]: I'd forgotten that you'd added them as well :). Will do.

**Comment by kpet** [comment]
Agree we need to address that as well. I've created https://github.com/KhronosGroup/OpenCL-CTS/issues/885.

File: test_common/harness/imageHelpers.cpp [hunk 2]
```diff
@@ -985,90 +922,6 @@ cl_ushort convert_float_to_half( float f )
...
-        return (cl_ushort)((int) x | sign);
-    }
-
-    u.u &= 0xFFFFE000U;
-    u.u -= 0x38000000U;
-
-    return (u.u >> (24-11)) | sign;
-}
-
 class TEST
```
Comment (jrprice) [inline test_common/harness/imageHelpers.cpp:925]: Can this class go now? It seems to just be testing the host functions for doing the conversions (and has it's own float->half conversion code in `__vstore_half_rte`).

Comment (kpet) [reply test_common/harness/imageHelpers.cpp]: Yeah, now removed.

**Pull request merged**